rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/HostsFileBasedInfrastructureManager.java

```diff
@@ -306,14 +310,63 @@ protected void addTimeouts(List<String> nodesUrl) {
         }
     }
 
+    protected void startNodeImplWithRetries(final InetAddress host, final int nbNodes, int retries)
+            throws RMException {
+        final List<String> depNodeURLs = new ArrayList<>(nbNodes);
+        try {
+            startNodeImpl(host, nbNodes, depNodeURLs);
+        } catch (Exception e) {
+            logger.warn("Failed nodes deployment in host : " + host + ", retries left : " + retries);
+            if (isInfiniteRetries(retries) || retries > 0) {
+                removeNodes(depNodeURLs);
+                waitPeriodBeforeRetry();
+                startNodeImplWithRetries(host, nbNodes, getRetriesLeft(retries));
```

Comment:
The infinite -1 parameter will actually not be infinite as it will hit a StackOverFlowError